src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/DatabaseChoiceListProvider.java

```diff
@@ -0,0 +1,419 @@
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_dbTable_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+        /**
+         * Validate a value inputed for driverdb
+         * 
+         * Checks followings:
+         * * not blank
+         * 
+         * @param driverdb
+         * @return FormValidation object
+         */
+        public FormValidation doCheckDriverdb(@QueryParameter String driverdb)
+        {
+            if(StringUtils.isBlank(driverdb))
+            {
+                return FormValidation.error(Messages.DbChoiceListProvider_jdbcDriver_empty());
+            }
+                       
+            return FormValidation.ok();
+        }
+        
+    }
+	
+	
+	/**
+	 * Recupere la liste des environnements recuperer de la base de donnees JACK	  
+	 *	 
+	 */
+	/**
+	 * 
+	 * @return
+	 */
+	private List<String> performDbQuery(){
+
+		Connection conn = null;
+		ResultSet resultat = null;
+		List<String> resultList = new ArrayList<String>();
+		try {
+			String driver = getJdbcDriver();
+			if (driver == null || StringUtils.isEmpty(driver)) {
+				LOGGER.log(Level.WARNING,"Invalid driver");
+			}
+			
+			Class.forName(driver);
+			
+			String user = getDbUsername();
+			if (user == null || StringUtils.isEmpty(user)) {
+				LOGGER.log(Level.WARNING,"Invalid user");
+			}
+
+			String password = getDbPassword();
+			if (password == null || StringUtils.isEmpty(password)) {
+				LOGGER.log(Level.WARNING,"Invalid password");
+			}
+
+			String urlBase = getJdbcUrl();
+			if (urlBase == null || StringUtils.isEmpty(urlBase)) {
+				LOGGER.log(Level.WARNING,"Invalid JDBC URL");				
+			}
+			
+			String colomndb = getDbColumn();
+			if (colomndb == null || StringUtils.isEmpty(colomndb)) {
+				LOGGER.log(Level.WARNING,"Invalid column");
+			}
+			
+			String namedb = getDbName();
+			if (namedb == null || StringUtils.isEmpty(namedb)) {
+				LOGGER.log(Level.WARNING,"Invalid DB name");
+			}
+
+			String tabledb = getDbTable();
+			if (tabledb == null || StringUtils.isEmpty(tabledb)) {
+				LOGGER.log(Level.WARNING,"table database invalid");				
+			}
+			
+			conn = DriverManager.getConnection(urlBase, user, password);
+						
+			// By default, a SELECT * is performed
+			String selectQuery = "select " + colomndb + " from " + namedb + "." + tabledb;
+						
+			// Use plain old JDBC to build and execute the query against the configured db
+			Statement statement = conn.createStatement();
+			boolean result = statement.execute(selectQuery);
+			if(result){
+				resultat = statement.executeQuery(selectQuery);
+				while(resultat.next()) {
+					resultList.add(resultat.getString(1));				  
+				}
+			}else{
+				LOGGER.log(Level.WARNING,"No result found with the query: " + selectQuery);
+			}
+		}
+		catch (SQLException se) {
+			LOGGER.log(Level.SEVERE,"Unable to access the database: " + dbName + "." + se);			
```

Comment:
Let's use [`Logger.log(Level, String, Throwable)`](http://docs.oracle.com/javase/6/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)).
It displays also stacktraces.